db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -926,11 +1389,279 @@ public CostEstimate estimateCost(OptimizablePredicateList predList,
             else
                 finalCostEstimate = secondPassCostEstimate;
         }
+
         return finalCostEstimate;
     }
 
+    private void reduceEstimatedCosts(AccessPath accessPath) {
+        accessPath.setCostEstimate(accessPath.getCostEstimate().cloneMe());
+
+        double costScaleFactor = 1e-9;
+        CostEstimate estimate = accessPath.getCostEstimate();
+
+        estimate.setLocalCost(estimate.getLocalCost() * costScaleFactor);
+        estimate.setRemoteCost(estimate.getRemoteCost() * costScaleFactor);
+        estimate.setLocalCostPerParallelTask(estimate.getLocalCostPerParallelTask() * costScaleFactor);
+        estimate.setRemoteCostPerParallelTask(estimate.getRemoteCostPerParallelTask() * costScaleFactor);
+    }
```

Comment:
If we could implement the fake UIS cd idea, then this is not needed. We simply add one more condition in deciding better plan to check whether we favor UIS code path. The code would be much less hacky, I think.

Reply:
Maybe we can discuss this as well.

Reply:
Since the access path enumeration question is answered, I think it's OK to do so. However, I feel a bit strange about reducing cost with a hard-coded factor. Is it possible to cost uis access path in a way that it's a fair comparison with costs of other access paths?

Reply:
I think the point is we don't want it to be fair.  When splice.optimizer.favorUnionedIndexScans is set, we want UIS to be picked all the time.  Decreasing all of the UIS access paths by the same factor would mean only another UIS access path with a similarly scaled cost estimate would beat the current one.  I'm not sure how else to get this same effect.